### PR TITLE
build: remove extra.bento-server dependency, add extra.tracing

### PR DESCRIPTION
**Pull request by parano**
## Checklist:

- [ ] My code follows the bentoml code style, both `./dev/format.sh` and
  `./dev/lint.sh` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: requirements/tests-requirements.txt [hunk 1]
```diff
@@ -5,6 +5,7 @@ coverage>=4.4
 flake8>=3.8.2
 flake8-length>=0.2.0
 isort>=5.0.0
+pydantic
```
Comment (parano) [inline requirements/tests-requirements.txt:8]: pydantic is required for running unit test

File: scripts/ci/config.yml [hunk 2]
```diff
@@ -11,27 +11,19 @@ unit:
   root_test_dir: "tests/unit"
   is_dir: true
   type_tests: "unit"
-  dependencies:
```
Comment (parano) [inline scripts/ci/config.yml:14]: these are already included in the tests-requirements.txt

File: scripts/ci/config.yml [hunk 3]
```diff
@@ -124,8 +114,6 @@ paddle:
   override_name_or_path: paddle
   dependencies:
     - "opencv-python==4.5.4.58"
-    - "pandas"
-    - "numpy"
```
Comment (parano) [inline scripts/ci/config.yml:128]: numpy is already BentoML's own dependency

File: codecov.yml [hunk 4]
```diff
@@ -69,7 +74,11 @@ flags:
...
-  bento-server:
+  e2e:
+    carryforward: true
+    paths:
+      - ".*"
+  unit:
     carryforward: true
     paths:
-      - "bentoml/_internal/**/*"
+      - ".*"
```
Comment (aarnphm) [inline codecov.yml:84]:
```suggestion
      - "bentoml/**"
```

File: codecov.yml [hunk 5]
```diff
@@ -69,7 +74,11 @@ flags:
       - bentoml/tensorflow.py
       - bentoml/transformers.py
       - bentoml/xgboost.py
-  bento-server:
+  e2e:
+    carryforward: true
+    paths:
+      - ".*"
```
Comment (aarnphm) [inline codecov.yml:80]:
```suggestion
      - "bentoml/**"
```

File: setup.cfg [hunk 6]
```diff
@@ -37,11 +37,11 @@ setup_requires =
 install_requires =
     click>=7.0
     simple-di>=0.1.2
-    attrs>=20
+    attrs>=21
```
Comment (parano) [inline setup.cfg:40]: This should be 20

**Pull request merged**